### DISCUSSION TITLE
Enable environment variable or hide default value

### DISCUSSION THREAD

**Discussion by BobVanB** [erikhuda/thor#841] [Ideas]
I would like to pass default passwords from environment variables and i don't want to see them in the generated help!

**What is happening:**
```
    class_option :rancher_secret_key,
                 :type     => :string,
                 :required => false,
                 :hide     => false,
                 :default  => ENV['RANCHER_SECRET_KEY'],
                 :banner   => "<rancher_secret_key>",
                 :desc     => "Default: ${RANCHER_ACCESS_KEY}"
```
output =>
```
Options:
  [--rancher-secret-key=<rancher_secret_key>]  # Default: ${RANCHER_ACCESS_KEY}
                                               # Default: asgraweihrasjkfhn378924019hfsadjfha78wrh
```
**What i would like to have:**
```
    class_option :rancher_secret_key,
                 :type     => :string,
                 :required => true,
                 :hide     => false,
                 :env      => "RANCHER_ACCESS_KEY"
                 :banner   => "<rancher_secret_key>",
                 :desc     => "Default: ${RANCHER_ACCESS_KEY}"
```
output => 
```
Options:
  --rancher-secret-key=<rancher_secret_key>    # Default: ${RANCHER_ACCESS_KEY}
```
This will results in the following behaviour:

1. the contents of `ENV["RANCHER_ACCESS_KEY"]` is hidden
2. if `param` is not given, the `ENV` is used
3. if both are nog given, the boolean for required is activated resulting in an error

**Other solution that could solve my problem**
Hide the default value!
```
    class_option :rancher_secret_key,
                 :type     => :string,
                 :required => true,
                 :hide     => false,
                 :hide_default => true,
                 :default => ENV["RANCHER_ACCESS_KEY"],
                 :banner   => "<rancher_secret_key>",
                 :desc     => "Default: ${RANCHER_ACCESS_KEY}"
```
output => 
```
Options:
  --rancher-secret-key=<rancher_secret_key>    # Default: ${RANCHER_ACCESS_KEY}
```

**Comment by BobVanB** [comment]
@rafaelfranca

Hi, can you review this issue and make a choice?  
That way i can develop towards that choice. 

With kind regards,

Bob

**Comment by rafaelfranca** [comment]
I think it makes sense. Sorry for the delay on replying to this, but if you are still interested in implementing it would be great.

**Comment by BobVanB** [comment]
Goodmorning @rafaelfranca , i'm not interested thank you. We choose a different library that was matching our needs. This was something i stumbled  on during review.
